docs/explanation/security/cryptographic-approaches.rst

```diff
@@ -0,0 +1,144 @@
+Cryptographic approaches in MicroCeph
+=======================================
+
+MicroCeph is a Ceph cluster in a single snap package. The snap is built on top of Ubuntu Ceph packages,
+and, therefore, shares some of their security features. This section makes references to `Cryptography in Ubuntu Ceph packages
+<https://ubuntu.com/ceph/docs/cryptographic-technologies-in-charmed-ceph`_.
+
+Snap security features
+----------------------
+
+MicroCeph is distributed as a snap package. Snaps offer some inherent security features, including:
+
+* **Sandboxing and confinement**: snaps are containerized applications that run in a sandboxed environment.
+  This isolation is achieved using Linux kernel security features such as ``AppArmor``, ``Seccomp``, and ``cgroups``.
+  These tools limit the system resources that snaps can access, preventing unauthorized access to the host system.
+
+* **Confinement level**: MicroCeph uses strict confinement. This level provides the highest security by restricting
+  access to system resources unless explicitly allowed through interfaces.
+
+* **Interfaces and plugs**: snaps use interfaces to request specific permissions for accessing system resources.
+  This allows for granular access to underlying hosts resources, i.e. only the resources a specific application
+  requires need to be allowed. Specific resources that can be used by MicroCeph are discussed in the next section.
+
+* **Updates**: snaps offer an easy network-based update mechanism. By default, snaps automatically upgrade;
+  however a best practice for MicroCeph production deployments is to hold automatic upgrades so that operators
+  can make use of zero-downtime upgrades for multi-node clusters.
+
+* **Cryptographic signatures**: the snap store uses cryptographic signatures to verify the integrity and authenticity of snaps.
+  This ensures that users download genuine software that has not been tampered with.
+
+* **Secure distribution**: The snap store acts as a central repository where snaps are published and distributed securely.
+  The store implements both automated checks and manual reviews to ensure the quality and security of the software it hosts.
+
+Resources used by MicroCeph snaps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interfaces
+^^^^^^^^^^
+
+MicroCeph snaps can use the following snap interfaces:
+
+* ``kernel-module-load``: to enable loading of Ceph-specific kernel modules, such as the ``rbd`` module.  
+* ``content``: to expose Ceph configuration to cooperating snaps.
+
+Plugs
+^^^^^
+
+MicroCeph snaps can use the following snap plugs:
+
+* ``block-devices``: for storage  
+* ``dm-crypt``: for disk encryption  
+* ``hardware-observe``: block device detection  
+* ``home``: access to user-supplied configuration and certificates  
+* ``microceph-support``: for additional storage  
+* ``mount-observe``: storage management  
+* ``network``: networking client  
+* ``network-bind``: network servers
```

Comment:
Meanwhile added another plug, `process-control`, to support resource limits config

Reply:
Done.